packages/site/site-nextjs/src/sitePreview/appRouter/sitePreviewRoute.ts

```diff
@@ -40,14 +40,19 @@ export async function sitePreviewRoute(request: NextRequest) {
  * @param options.skipDraftModeCheck Allows skipping the draft mode check, only required when called from middleware.ts (see https://github.com/vercel/next.js/issues/52080)
  * @return If SitePreview is active the current preview settings
  */
-export async function previewParams(options: { skipDraftModeCheck: boolean } = { skipDraftModeCheck: false }): Promise<SitePreviewParams | null> {
+export async function previewParams(options: { skipDraftModeCheck: boolean } = { skipDraftModeCheck: false }): Promise<PreviewParams | null> {
     if (!options.skipDraftModeCheck) {
         if (!draftMode().isEnabled) return null;
     }
 
-    const cookie = cookies().get("__comet_preview");
-    if (cookie) {
-        return verifySitePreviewJwt(cookie.value);
+    const sitePreviewCookie = cookies().get("__comet_preview");
+    if (sitePreviewCookie) {
+        return verifyJwt<PreviewParams>(sitePreviewCookie.value);
     }
+    const blockPreviewCookie = cookies().get("__comet_block_preview");
+    if (blockPreviewCookie) {
+        return verifyJwt<PreviewParams>(blockPreviewCookie.value);
+    }
```

Comment:
Although you have two separate cookies, the site preview always has precedence. That means, if the user once has opened the site preview (that stores the cookie for 24h) the block preview will always use that scope.

(Same problem with multiple browser tabs that each has a block preview with different scope opened. But that is a minor issue that will not happen that often in practice)

Possible solution: use only a single cookie? (still breaks when site preview + block preview is opened in two browser tabs)

Reply:
Or, maybe the cookie approach isn't the way to go (because of these issues)